### PR TITLE
Merge/release 3.1.1 master

### PR DESCRIPTION
**Pull request by alexandruatomei9**

### CODE REVIEW

File: src/main/java/org/mule/modules/cassandradb/internal/service/CassandraServiceImpl.java [hunk 1]
```diff
@@ -103,13 +105,17 @@ public void changeColumnType(String tableName, String keyspaceName, AlterColumnI
...
     public void insert(String keyspaceName, String table, Map<String, Object> entity) {
         String keyspace = getKeyspaceNameToUse(keyspaceName);
-
         Insert insertObject = QueryBuilder.insertInto(keyspace, table);
+        List<ColumnMetadata> columns = new CassandraMetadata(getConnection()).getTableMetadata(keyspaceName, table).getColumns();
 
         for (Map.Entry<String, Object> entry : entity.entrySet()) {
-            insertObject.value(entry.getKey(), entry.getValue());
+            String columnType = DataTypeResolver.getColumnType(columns, entry.getKey());
+            if (columnType.equals("timestamp") || columnType.equals("time")) {
```
Comment (danpece) [inline src/main/java/org/mule/modules/cassandradb/internal/service/CassandraServiceImpl.java]: please reverse the condition to avoid NPEs

Comment (danpece) [reply src/main/java/org/mule/modules/cassandradb/internal/service/CassandraServiceImpl.java]: or use this approach:

// Define a constant in your class. Use a HashSet for performance
private static final Set<Integer> values = new HashSet<Integer>(Arrays.asList(12, 16, 19));

// In your method:
if (values.contains(x)) {
    ...
}

File: src/main/java/org/mule/modules/cassandradb/internal/service/CassandraServiceImpl.java [hunk 2]
```diff
@@ -122,9 +128,15 @@ public void update(String keySpace, String table, Map<String, Object> entityToUp
             throw new QueryErrorException("Mismatched input. SET and WHERE clause must not be null.");
         }
         Update updateObject = QueryBuilder.update(keySpace, table);
+        List<ColumnMetadata> columns = new CassandraMetadata(getConnection()).getTableMetadata(keySpace, table).getColumns();
 
         for (Map.Entry<String, Object> entry : entity.entrySet()) {
-            updateObject.with().and(QueryBuilder.set(entry.getKey(), entry.getValue()));
+            String columnType = DataTypeResolver.getColumnType(columns, entry.getKey());
+            if (columnType.equals("timestamp") || columnType.equals("time")) {
```
Comment (danpece) [inline src/main/java/org/mule/modules/cassandradb/internal/service/CassandraServiceImpl.java]: same as above

File: src/main/java/org/mule/modules/cassandradb/internal/util/DataTypeResolver.java [hunk 3]
```diff
@@ -32,4 +40,50 @@ public static DataType resolve(ColumnType type){
             default: return DataType.text();
         }
     }
+
+    public static Date resolveToDate(String date) {
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
```
Comment (danpece) [inline src/main/java/org/mule/modules/cassandradb/internal/util/DataTypeResolver.java]: extract this as static final field 
(if so; please note that SimpleDateFormat class is not thread safe; so resoveToDate should be synchronized in this case)

File: src/main/java/org/mule/modules/cassandradb/internal/util/DataTypeResolver.java [hunk 4]
```diff
@@ -32,4 +40,50 @@ public static DataType resolve(ColumnType type){
...
+            dateResolved = dateFormat.parse(date);
+            return dateResolved;
+        } catch (ParseException e) {
+            throw new QueryErrorException("Error converting to date format", e);
+
+        }
+    }
+
+    public static boolean isValidDate(String date) {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
```
Comment (danpece) [inline src/main/java/org/mule/modules/cassandradb/internal/util/DataTypeResolver.java]: same as above

File: src/main/java/org/mule/modules/cassandradb/internal/util/DataTypeResolver.java (hunk 3 above)
Comment (alexandruatomei9) [reply src/main/java/org/mule/modules/cassandradb/internal/util/DataTypeResolver.java]: done

File: src/main/java/org/mule/modules/cassandradb/internal/util/DataTypeResolver.java (hunk 4 above)
Comment (alexandruatomei9) [reply src/main/java/org/mule/modules/cassandradb/internal/util/DataTypeResolver.java]: done

File: src/main/java/org/mule/modules/cassandradb/internal/util/DataTypeResolver.java (hunk 3 above)
Comment (alexandruatomei9) [reply src/main/java/org/mule/modules/cassandradb/internal/util/DataTypeResolver.java]: only extracted the string representing the format. no synchronization needed

File: src/main/java/org/mule/modules/cassandradb/internal/service/CassandraServiceImpl.java (hunk 2 above)
Comment (alexandruatomei9) [reply src/main/java/org/mule/modules/cassandradb/internal/service/CassandraServiceImpl.java]: done

File: src/main/java/org/mule/modules/cassandradb/internal/service/CassandraServiceImpl.java (hunk 1 above)
Comment (alexandruatomei9) [reply src/main/java/org/mule/modules/cassandradb/internal/service/CassandraServiceImpl.java]: done

**Pull request merged**